README-dev-on-docker.md

```diff
@@ -0,0 +1,114 @@
+# Developing Vespa on Docker
+
+The purpose of this document is to describe how to set up a combined development and
+run-time environment for (iteratively) developing and testing Vespa.
+
+### Start the docker container
+
+    DOCKER_IMAGE=vespaengine/vespa-dev:latest
+    SOURCE_ROOT=${HOME}/vespa/vespa
+    APP_ROOT=${HOME}/vespa/sample-apps
+
+    docker run -d -p 8080:8080 -p 8998:8998 -p 10001:10001 \
+                -v ${HOME}/.ccache:/root/.ccache:delegated \
+                -v ${HOME}/.m2:/root/.m2:delegated \
+                -v ${SOURCE_ROOT}:/source:delegated \
+                -v ${APP_ROOT}:/apps:delegated \
+                --privileged \
+                --name vespa \
+                ${DOCKER_IMAGE} \
+                /bin/sh -c "tail -f /dev/null"
+
+Change the above paths and container name to reflect your system.
+
+The ports defined here are for connecting to the Vespa service from the host
+machine. 8080 is the default HTTP port for querying and feeding to Vespa. The
+other ports are optional, the ones here are for Java remote debugging and
+YourKit profiling.
+
+We map in multiple directories from the host to the docker container. They
+are all mapped with `:delegated` which speeds up writing to disk in the container. This
+is helpful for instance for compiling.
+
+The first directory is the `.ccache` directory. Having this directory on the host
+system significantly speeds up C++ compiling. The first C++ build will be slow, but
+subsequent builds will greatly benefit from this cache.
+
+The `.m2` directory is the maven repository, and if Java is built in the container
+artifacts will be populated to the host repository as well.
+
+The source root is the Vespa source code, and the apps root is the root
+directory for any Vespa applications you would like to try out. Here it is set
+to the sample-apps directory.
+
+### Build code
+
+Vespa has both Java and C++ code. You can build Java either on the host or inside
+the container, but host building will typically be faster.
+
+    $ ./bootstrap.sh java
```

Comment:
building outside the container assumes a lot about my setup. If we want to support this we should link to a more detailed description of the needed Java environment. (this just hangs for me)

Reply:
hangs inside container also, probably due to mounting .m2 directory...

Reply:
does not work without or with empty .m2 either. I think we need settings.xml, maybe?

Reply:
@havardpe This was due to bootstrap.sh not offering any output while downloading artifacts, right?

Reply:
seems to just be extremely slow due to quiet mvn downloading the internet, my bad